#170972171 Signup using database

Muhire-Josue opened the pull request:
#### What does this PR do?

- User can sign up in the database

#### Description of Task to be completed?

- It uses postgres database to store user information

#### How should this be manually tested?

- Use this link in postman: https://anounce-it.herokuapp.com/api/v1/auth/signup

#### Any background context you want to provide?

- N/A

#### What are the relevant pivotal tracker stories?

- [170972171](https://www.pivotaltracker.com/story/show/170972171)

#### Screenshots (if appropriate)
![signup](https://user-images.githubusercontent.com/37122177/73361066-e1e65c00-42ac-11ea-936b-00db7a82b646.PNG)
![user-duplication](https://user-images.githubusercontent.com/37122177/73361068-e1e65c00-42ac-11ea-8db4-33ed30cc1d59.PNG)
![welcome](https://user-images.githubusercontent.com/37122177/73361069-e1e65c00-42ac-11ea-88f5-320ed5578763.PNG)

Muhire-Josue committed:
feature(signup): add signup using database

- add signup using database
- add tests
- add validations

[Finishes #170972171]

Muhire-Josue committed:
featurea(sign up): add sign up using database

- add fix path to tests for travis
[Finishes #170972171]

j4l13n commented on .babelrc:
```diff
@@ -1,3 +1,12 @@
+            }
+        ]
+    ]
 }
```
Fix this new line issue

j4l13n commented on server/Tests/userTEST.test.js:
```diff
@@ -45,48 +45,47 @@ describe('User tests', () => {
         done();
       });
   });
+  // it('Should login a user', (done) => {
```
Remove the commented code before you proceed

j4l13n reviewed (commented):
Good job @Muhire-Josue, Fix all small issues that you have before you merge this PR

Muhire-Josue committed:
feature(sign up): add sign up

- add fix new line issue in babelrc file
- add removed all commented lines from userTest
[Finishes #170972171]

j4l13n commented on server/Routes/userRoutes.js:
```diff
@@ -7,7 +7,7 @@ import userFound from '../Middlewares/userFound';
+router.get('/api/v2/', userController.welcomeToAPI);
+router.post('/api/v2/auth/signup', userValidation, userDuplication, userController.signUp);
 router.post('/api/v1/auth/signin', userFound, incorrectPassword, userController.login);
 export default router;
```
Fix this as well

Muhire-Josue replied:
Should I leave it to v1 instead of v2?

j4l13n reviewed (commented):
Fix line issues

Muhire-Josue committed:
feature(sign up): add sign up using database

- add fix an issue in userRoute
[Finishes #170972171]
